app/controllers/spree/admin/reports_controller_decorator.rb

```diff
@@ -250,23 +215,29 @@ def xero_invoices
     @distributors = Enterprise.is_distributor.managed_by(spree_current_user)
     @order_cycles = OrderCycle.active_or_complete.accessible_by(spree_current_user).order('orders_close_at DESC')
 
-    @report = OpenFoodNetwork::XeroInvoicesReport.new spree_current_user, params
+    @report = OpenFoodNetwork::XeroInvoicesReport.new spree_current_user, params, render_content?
     render_report(@report.header, @report.table, params[:csv], "xero_invoices_#{timestamp}.csv")
   end
 
+  private
+
+  # We don't want to render data unless search params are supplied.
+  # Compiling data can take a long time.
+  def render_content?
+    request.post?
```

Comment:
Yeah, I was wondering about that. My first draft looked for common search parameters. The question is: how difficult is it to maintain this list? Do we look for the presence or absence of search parameters?

One difficulty is that some actions change the `params` object. They set default values. So I need to determine the search state before that. Or I change the code to not change the `params` object, because I think it's bad practice. But Spree's own implementation does that:

https://github.com/spree/spree/blob/baadc8f95cbd25ff8b1741742dbaed75045cbc25/backend/app/controllers/spree/admin/reports_controller.rb#L29